src/common/url/construct-url.ts

```diff
@@ -0,0 +1,4 @@
+export const constructUrlCurrentHref = (searchParams: string): string => {
+  const base = window.location.href.split("?")[0];
+  return searchParams ? base + "?" + searchParams : base;
+};
```

Comment:
This should probably be an `add` function? What if the current URL already has a search param?

Reply:
```ts
const urlParams = new URLSearchParams(window.location.search);
  Object.entries(params).forEach(([key, value]) => {
    urlParams.append(key, value);
  });
`${window.location.origin}${window.location.pathname}?${urlParams.toString()}`
```

Reply:
But then in `search-params.ts`?

Reply:
in `search-params`